pandas/_libs/intervaltree.pxi.in

```diff
@@ -97,6 +99,27 @@ cdef class IntervalTree(IntervalMixin):
             self._right_sorter = np.argsort(self.right)
         return self._right_sorter
 
+    @property
+    def is_overlapping(self):
+        """
+        Determine if the IntervalTree contains overlapping intervals.
+        """
+        if self._is_overlapping is not None:
+            return self._is_overlapping
+
+        # <= when both sides closed since endpoints can overlap
+        op = le if self.closed == 'both' else lt
+
+        self._is_overlapping = False
+        for previous, current in zip(self.left_sorter, self.left_sorter[1:]):
+            # overlap if start of current interval < end of previous interval
```

Comment:
can't this just be an array comparison?

Reply:
Yes, but wouldn't that generally be a bit slower?  For the array comparison it'd be something like:
```python
(self.left[self.left_sorter[1:]] < self.right[self.left_sorter[:-1]]).any()
```
Wouldn't the interior comparison of `self.left[...] < self.right[...]` be computed in it's entirety before the `any` is evaluated?  Whereas the elementwise way it's currently written would terminate after the first successful `<` evaluation?  Or is there something I'm missing here?

Reply:
yes but it’s worth testing
as it’s a bit more idiomatic to use array slicing here and it might be faster

Reply:
Looks like you're right.  I did some timings with some non-cached versions of the loop and array methods (still cached the sorters since both methods use them in the same way), aptly named `is_overlapping_loop` and `is_overlapping_array`.  Opted to switch to the array version since it's speed is consistent, and faster in a wider variety of scenarios.

When the overlap occurs at the beginning, the loop version is the fastest (pretty obvious):
```python
In [2]: ii = pd.IntervalIndex.from_arrays(
   ...:     np.arange(0, 2*10**5), np.arange(2, 2*10**5 + 2))

In [3]: ii._engine.left_sorter; ii._engine.right_sorter; pass  # cache sorters

In [4]: %timeit ii._engine.is_overlapping_loop
980 ns ± 19.1 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [5]: %timeit ii._engine.is_overlapping_array
2.06 ms ± 66.2 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
``` 

If no overlaps are present, and both methods need to do full passes, the array method is faster:
```python
In [6]: ii = pd.interval_range(0, 2*10**5)

In [7]: ii._engine.left_sorter; ii._engine.right_sorter; pass  # cache sorters

In [8]: %timeit ii._engine.is_overlapping_loop
80.8 ms ± 6.35 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [9]: %timeit ii._engine.is_overlapping_array
2.19 ms ± 156 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

If an overlap occurs midway through the array method is still faster, with the loop method taking about half the time as the non-overlapping version (makes sense):
```python
In [10]: left = list(range(10**5)) + [10**5 - 2] + list(range(10**5, 2*10**5))
    ...: right = list(range(1, 10**5 + 1)) + [10**5 + 2] + list(range(10**5+1, 2*10**5+1))
    ...: ii = pd.IntervalIndex.from_arrays(left, right)

In [11]: ii._engine.left_sorter; ii._engine.right_sorter; pass  # cache sorters

In [12]: %timeit ii._engine.is_overlapping_loop
39.6 ms ± 1.2 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

In [13]: %timeit ii._engine.is_overlapping_array
2.23 ms ± 127 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```